### PR TITLE
Enabling es6 modules for the packages in the repo

### PR DESCRIPTION
**Pull request by dzearing**
Webpack 4 is being prepped. As part of that, there are some significant gains in its ability to tree shake out things not referenced. This however requires a few pre-requisites:

1. The library must output es6 modules.
2. The package.json must have a "module" entry point.
3. Assuming the package has no global code that should be in the bundle by default, it must specify `"side-effects": false` in its package.json
4. Import statements to other libraries need to be package imports (e.g. `@uifabric/utilities`) and not file based imports (e.g `@uifabric/utilities/lib/index`). This will give webpack the freedom to choose the `lib-es6/index` over the `lib/index` file.

This change addresses all of these things. 

### Bundling currently

Let's say we bundle the `Fabric` component. Today, you would bundle it from the top level import:

index.ts:
```tsx
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';

ReactDOM.render(<Fabric/>, document.body);
```
Doing this with webpack 3 today, you would see a gzip size of about 20.3k.

image

### Bundling with webpack 4 and these changes

Now with tree shaking working, you no longer need to import from a random /lib/* top level import; you just import from the package giving you full intellisense for anything you want to use:

index.ts:
```tsx
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Fabric } from 'office-ui-fabric-react';

ReactDOM.render(<Fabric/>, document.body);
```

Webpack 4 will traverse the package and only include the bits that the "Fabric" component would need. The result drops down to 3.7k gzipped.

image

### CODE REVIEW

File: packages/office-ui-fabric-react/src/index.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export * from './Utilities';
 // Using the default import, include all icon definitions. Products that care
 // about bundle size should not be using the main entry, until tree shaking
 // is perfected. (Use the top level imports instead.)
-import { initializeIcons } from '@uifabric/icons/lib/index';
+import { initializeIcons } from '@uifabric/icons';
 
 // TODO: remove the @beta tag once they're publised to the CDN.
```
Comment (elliottsj) [inline packages/office-ui-fabric-react/src/index.ts]: Can this TODO comment be removed now?

Comment (dzearing) [reply packages/office-ui-fabric-react/src/index.ts]: Yes, thanks!

**Comment by dzearing** [comment]
I realized that webpack is not including anything from @uifabric/utilities. I don't understand why. Investigating more deeply.

**Comment by dzearing** [comment]
Just an fyi on this; i'm keeping this around and will come back to it as webpack 4 stabilizes a bit. Need to update the `side-effects` to be `sideEffects`

**Comment by dzearing** [comment]
We will re-open this in the future, but until Webpack4 is released it would just slow the build down.

**Pull request closed without merging**